### PR TITLE
React workflow fixes

### PR DESCRIPTION
**Pull request by daileytj**
Fixes PXBLUE-2255, #72, and #67.

#### Changes proposed in this Pull Request:
- Allow users to go back during all steps in the account registration workflows
- Changed EULA page order so it shows first during self registration
- Add ability to customize character limits for first and last name text fields
- Clear Invalid credentials error message after editing input

** Use shared-auth branch `feature/add-name-character-limits` to test these.

To test the custom character limit, uncomment the `registrationConfig` definition in the `AuthUIContextProvider` in the example project's `App.tsx` file.

To test invalid credentials error message, edit the example project's `AuthUIActions.tsx` login method to immediately return `throw new Error('LOGIN.INVALID_CREDENTIALS');`. Start up example project and try to login. The "Invalid credentials" error should display on the email and password inputs. Modify either input and observe the error being removed until submitting and receiving another transit error.

### CODE REVIEW

File: login-workflow/src/screens/Login.tsx [hunk 1]
```diff
@@ -334,7 +335,10 @@ export const Login: React.FC = () => {
                         label={t('pxb:LABELS.PASSWORD')}
                         className={clsx(classes.passwordFormField, { [classes.hasError]: isInvalidCredentials })}
                         value={passwordInput}
-                        onChange={(evt: ChangeEvent<HTMLInputElement>): void => setPasswordInput(evt.target.value)}
+                        onChange={(evt: ChangeEvent<HTMLInputElement>): void => {
+                            setPasswordInput(evt.target.value);
+                            authUIState.login.transitErrorMessage = null;
```
Comment (joebochill) [inline login-workflow/src/screens/Login.tsx:340]: this works, but it feels like it's probably not a good practice to directly mutate the value that we are pulling from a Context. I think a safer bet would be to dispatch the reset action which will fully clear the TransitState:
```tsx
if(authUIState.login.transitErrorMessage !== null) AccountActions.resetLogin();
```

**Review by joebochill — CHANGES_REQUESTED** [review]
I ran into a bunch of issues getting the demo to run, but finally finagled it to work. You may want to mention in the testing instructions that you need to use the specific branch of the shared auth submodule. And even then the linking script needs to be updated so that it links the shared auth code to the workflow in order for the workflow to build successfully. Long story short...we probably need to come in and clean that up a bit at some point.

It gets really hairy when you commit changes to the submodule in the parent repository — we should be avoiding this like the plague (it's easy to accidentally do it when you do something like git add -A). The only time we should be updating the pointer is when new changes have been merged into the dev branch of the submodule and we want to pick those up using 'yarn update:submodule'.

Everything looks good on a visual inspection except for:
- weird loading animation in self registration after the EULA screen - nothing should be happening between these screens so it's very odd to see the loading effect
- In self registration if you go EULA -> Next -> Back, you are on the wrong screen

For the second point above, it looks like you changed the order of the screens, but didn't make any changes to some of the logic below that changes screens based on the screen you are on. Have a look at the useEffect on line 383 (this should be happening on the enter email screen now instead of EULA), and the advance logic on line 414.

Do we want to make the back button say 'Cancel' on the EULA screen for invite-registration like we're doing for Angular (I assume yes)?
```tsx
{isFirstStep ? t('pxb:ACTIONS.CANCEL') : t('pxb:ACTIONS.BACK')}
```

**Review by joebochill — APPROVED** [review]
Can you re-update the shared auth submodule pointer now that the dev branch of the shared-auth repo has your latest updates?

Question for @huayunh — do we want the back button to say cancel on the first screen of the Self-Registration flow too? For consistency?

Changes look good 👍

**Comment by joebochill** [comment]
We may want to hold off on fully publishing this and the RN one, just until I hear back from the WiFi wiring team...they're determining if we should be allowed to go back after validating the email or not. If PX white invalidates the code after it is checked then we will have to block the back logic on that one step (I don't think they do that, but just want to be sure).

**Comment by huayunh** [comment]
> Question for @huayunh — do we want the back button to say cancel on the first screen of the Self-Registration flow too? For consistency? 

yes, that would be nice.

**Comment by joebochill** [comment]
> > Question for @huayunh — do we want the back button to say cancel on the first screen of the Self-Registration flow too? For consistency?
> 
> yes, that would be nice. Or even better:

We could do that for RN. It would feel weird to me to have a close button like that on web.

**Comment by daileytj** [comment]
@huayunh @joebochill Just to clarify, for RN we want to hide the back/cancel button for the first screen in both registration pagers and for React, we're just changing the button text to say "cancel" on the first screen for both registration pagers?

**Comment by joebochill** [comment]
I think that is what we are saying.

On RN there should be a CloseHeader component that you can make use of I think.

**Pull request merged**